### ISSUE TITLE
Feature request: URI Validation method that returns boolean

### ISSUE THREAD

**Issue by benSlaughter** [sporkmonger/addressable#257]
Having come across an issue using ruby-json-schema, a discussion has been started [here](https://github.com/ruby-json-schema/json-schema/pull/373) regarding parsing a URI using addressable, and managing failures and flow control with exceptions.
It was mentioned that having a method similar to `parse` that would validate a URI and return a boolean value if the URI was valid instead of raising an exception.

tl;dr Can we have a URI validation method please that returns a boolean value.

**Comment by dentarg** [comment]
> having a method similar to parse that would validate a URI and return a boolean value if the URI was valid instead of raising an exception

This is actually what the gem [twingly-url](https://github.com/twingly/twingly-url) is doing, see https://github.com/twingly/twingly-url#getting-started for some examples. It might be useful for you, or give you some inspiration on how to wrap addressable yourself. twingly-url is using addressable and [public_suffix](https://github.com/weppos/publicsuffix-ruby) under the covers. (Full disclosure: I work for the company that made twingly-url)

**Comment by benSlaughter** [comment]
Nothing wrong with plugging your own gems ;)

Thanks for the response, I'm not the owner of the code with the issue, I was enquiring as I liked the idea of a non aggressive validation method.
I'll pass this on 👍

**Comment by RST-J** [comment]
Of course we can handle addressable's execption and wrap it internally in json-schema to get along. But as using exceptions for control flow is considered bad in general in programming (at least this is my perception), I thought it would be worth to raise this issue here at the root. And really no harm done if @sporkmonger looks at it and decides to leave things as they are. Its only about giving it a thought.

**Comment by dentarg** [comment]
I'm going to close this one, and recommend use of https://github.com/twingly/twingly-url (or your own wrapper) if you want this. While it has a strange name (hehe) it does this and I think it does it well. I no longer work at Twingly but I still use it when I need to check if a URL is valid or not.
